### PR TITLE
Feat/dotnet agent windows

### PR DESCRIPTION
**Pull request by Julien4218**

### CODE REVIEW

**Review by Julien4218 — COMMENTED** [review]
@devfreddy and/or @jbeveland27 
Please review this PR for the .net agent install on windows, this would help demoing tomorrow.
I've just did a manual test on Staging and this works.
cc @jaffinito

File: recipes/newrelic/apm/.net/windows/net-install.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          }
+          $xdoc.Save($configPath)'
+        - echo "New Relic .NET agent configured"
+
+    start:
+      cmds:
+        - cmd /c "iisreset /start"
+
+    ensure_sites_started:
+      ignore_error: true
```
Comment (jbeveland27) [inline recipes/newrelic/apm/.net/windows/net-install.yml:79]: Just wondering, why are we ignoring errors for this task?

File: test/definitions/apm/.net/windows/windows2019-iis.json [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+          "id": "nr_infra",
+          "resource_ids": ["iiswindows2019"],
+          "provider": "newrelic",
+          "source_repository": "https://github.com/newrelic/open-install-library.git",
+          "deploy_script_path": "test/deploy/windows/newrelic-cli/install-recipe/roles",
+          "params": {
+              "recipe_content_url": [
+                  "https://raw.githubusercontent.com/newrelic/open-install-library/main/recipes/newrelic/apm/.net/windows/net-install.yml"
+              ]
+          }
```
Comment (jbeveland27) [inline test/definitions/apm/.net/windows/windows2019-iis.json:64]: Just food for thought: maybe have a separate test that installs both infra agent and .NET agent? I'm mostly curious to know if that would "just work".

File: recipes/newrelic/apm/.net/windows/net-install.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+          $file = Resolve-Path("$configPath");
+          $xdoc = New-Object System.Xml.XmlDocument;
+          $xdoc.PreserveWhitespace = $true;
+          $xdoc.Load($file);
+          $xdoc.configuration.application.RemoveAll();
+          if ("{{.NEW_RELIC_REGION}}" -like "STAGING"){
+            $xdoc.configuration.service.SetAttribute("host", "staging-collector.newrelic.com")
+          }
+          $xdoc.Save($configPath)'
+        - echo "New Relic .NET agent configured"
```
Comment (jbeveland27) [inline recipes/newrelic/apm/.net/windows/net-install.yml:72]: Should we offer a way to set the app name in the config file? (I saw the 3 apps had the right name in your screenshot, how did that work?)

**Review by jbeveland27 — APPROVED** [review]
Looks good, just have a few questions.

File: recipes/newrelic/apm/.net/windows/net-install.yml (hunk 3 above)
Comment (jaffinito) [reply recipes/newrelic/apm/.net/windows/net-install.yml:72]: By removing the name from the application element (line 67), the agent will revert to auto-naming via the app pool name.  As long as the virtuoso UI can ask for and provide the names (3 max), we can easily add them in the configure step.

File: recipes/newrelic/apm/.net/windows/net-install.yml (hunk 1 above)
Comment (Julien4218) [reply recipes/newrelic/apm/.net/windows/net-install.yml:79]: It's not critical for that task to work.

File: test/definitions/apm/.net/windows/windows2019-iis.json (hunk 2 above)
Comment (Julien4218) [reply test/definitions/apm/.net/windows/windows2019-iis.json:64]: Yes, once we clarified our valid installs path (only stitch path?).

File: recipes/newrelic/apm/.net/windows/net-install.yml (hunk 3 above)
Comment (Julien4218) [reply recipes/newrelic/apm/.net/windows/net-install.yml:72]: Correct, not part of this work. We use automatic names, this simplifies the work.

**Pull request merged**